src/pyenphase/auth.py

```diff
@@ -14,7 +14,7 @@ def __init__(self) -> None:
         """Initialize the EnvoyAuth class."""
         pass
 
-    async def setup(self) -> None:
+    async def setup(self, client: httpx.AsyncClient, host: str) -> None:
```

Comment:
Is it better to pass the host in the constructor or in the setup function?

Reply:
Probably better to do it in the constructor if we can